### PR TITLE
Statistics prerequisite refactorings

### PR DESCRIPTION
**Pull request by ArturGajowy**
Prerequisite refactorings for comparing cost estimates with actual costs.

@sopel39 @losipiuk please review

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/util/MoreMaps.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+package com.facebook.presto.util;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.BinaryOperator;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toMap;
+
+public class MoreMaps
```
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/util/MoreMaps.java]: final

File: presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanPrinter.java [hunk 2]
```diff
@@ -1476,8 +1476,8 @@ public static PlanNodeStats merge(PlanNodeStats left, PlanNodeStats right)
             long planNodeOutputPositions = left.planNodeOutputPositions + right.planNodeOutputPositions;
             DataSize planNodeOutputDataSize = succinctBytes(left.planNodeOutputDataSize.toBytes() + right.planNodeOutputDataSize.toBytes());
 
-            Map<String, OperatorInputStats> operatorInputStats = mergeOperatorInputStatsMaps(left.operatorInputStats, right.operatorInputStats);
```
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanPrinter.java]: shouldn't you remove `mergeOperatorInputStatsMaps` and
`mergeOperatorHashCollisionsStatsMaps`?

File: presto-main/src/main/java/com/facebook/presto/sql/planner/Plan.java [hunk 3]
```diff
@@ -16,22 +16,24 @@
...
-    public Plan(PlanNode root, SymbolAllocator symbolAllocator)
+    public Plan(PlanNode root, Map<Symbol, Type> types)
     {
         requireNonNull(root, "root is null");
-        requireNonNull(symbolAllocator, "symbolAllocator is null");
+        requireNonNull(types, "types is null");
 
         this.root = root;
-        this.symbolAllocator = symbolAllocator;
+        this.types = new HashMap<>(types);
```
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/Plan.java]: `ImmutableMap.copyOf`

File: presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanPrinter.java (hunk 2 above)
Comment (ArturGajowy) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanPrinter.java]: They're used as method references and I preferred them used that way.

**Comment by ArturGajowy** [comment]
Comments addressed + rebased. @martint please have a look, it's needed for #7212

**Comment by ArturGajowy** [comment]
@martint please have a look these are automated refactorings and extractions - easy to review and very hard to rebase in case of conflicts

**Comment by ArturGajowy** [comment]
@martint please have a look

**Comment by losipiuk** [comment]
@martint Commits from this PR are also present on statistics PR (https://github.com/prestodb/presto/pull/7212), but it would be best to merge these first.
This is just a bunch of automated refactorings which (we believe) are beneficial on their own and are prerequisites to some of the later commits in statistics PR.
When you merge this one I will remove the overlapping commits from statistics PR and we can resume the PR on that one.

It should be not hard to review this PR as it does not contain almost any logic changes. Just `Add tpch.column-naming property to the TPCH connector` is minor extension of tpch-connector. 
Could you (or someone) give it a look?

File: presto-main/src/main/java/com/facebook/presto/operator/ExplainAnalyzeOperator.java [hunk 4]
```diff
@@ -146,7 +146,7 @@ public Page getOutput()
             return null;
         }
 
-        String plan = textDistributedPlan(queryInfo.getOutputStage().get().getSubStages(), metadata, operatorContext.getSession());
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/operator/ExplainAnalyzeOperator.java]: Why do we need this change?

File: presto-main/src/main/java/com/facebook/presto/util/MoreMaps.java [hunk 5]
```diff
@@ -26,7 +26,12 @@ private MoreMaps() {}
 
     public static <K, V> Map<K, V> mergeMaps(Map<K, V> map1, Map<K, V> map2, BinaryOperator<V> merger)
     {
-        return Stream.of(map1, map2)
+        return mergeMaps(Stream.of(map1, map2), merger);
+    }
+
+    public static <K, V> Map<K, V> mergeMaps(Stream<Map<K, V>> mapStream, BinaryOperator<V> merger)
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/util/MoreMaps.java]: Where is this used?

File: presto-main/src/main/java/com/facebook/presto/operator/ExplainAnalyzeOperator.java (hunk 4 above)
Comment (ArturGajowy) [inline presto-main/src/main/java/com/facebook/presto/operator/ExplainAnalyzeOperator.java]: It's not strictly needed - it's a readability and contract improvement: that method seems to require a very specific list of `subStages` - the one from `outputStage`. Previously it was not clear which exactly `stageInfos` it needed. See [here](https://github.com/prestodb/presto/pull/7442/commits/c1254b4a969f63e7cfad106abde142cd4d773ca6#diff-11a661ee627130692b6f5c4ab02b1354R183) for the signature change.

File: presto-main/src/main/java/com/facebook/presto/util/MoreMaps.java (hunk 5 above)
Comment (ArturGajowy) [inline presto-main/src/main/java/com/facebook/presto/util/MoreMaps.java]: in the dependent #7212, in [this line](https://github.com/prestodb/presto/pull/7212/commits/ae283fca46deaa9ab0d91763d6a12b86c6edfec6#diff-943d7ae1ce365f0b04a2a7e5fdf17339R65)

**Comment by martint** [comment]
I merged everything (after squashing a few commits that should go together) except for "Introduce MoreMaps.mergeMaps(mapStream, mergerOperator)". I'll leave that one until I work on the dependent PR, as it doesn't make a whole lot of sense in isolation.

**Pull request closed without merging**